### PR TITLE
Feature: Block Custom Views

### PR DESCRIPTION
**Pull request by nielslyngsoe**
Finishes the implementation of custom views for Block Editors.

Only slightly breaking as if someone already did implement a Custom View, then things would have changed a little bit(we do not parse on a prop called `urls`. But the feature was never complete neither exactly working so I do not see it as breaking.

Notice, this also makes Extension Initializers call their callback method with an empty array if there is no Extensions that fits the filtering. Giving developers a response to react to, giving knowledge on when to fallback to other approaches than extensions. (This maybe be an even more important breaking change, but not really breaking for any of our implementations as it is perfectly fine to expect retrieving an empty array when nothing was permitted).
This as well has been used in the extension-slot, which now 

Fixes https://github.com/umbraco/Umbraco-CMS/issues/16134
Fixes https://github.com/umbraco/Umbraco-CMS/issues/16736

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## How to test?

Search for and remove out-commenting of this to have some custom component to test with:
```// TODO: Remove test custom view, or transfer to test or similar?```

Do correct the manifest of the element to fit your test case.

## Checklist

- [ ] If my change requires a change to the documentation, I have updated the documentation in this pull request.
- [ ] I have read the **[CONTRIBUTING](<(https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/main/.github/CONTRIBUTING.md)>)** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: src/packages/block/manifests.ts [hunk 1]
```diff
@@ -4,8 +4,11 @@ import { manifests as blockListManifests } from './block-list/manifests.js';
 import { manifests as blockRteManifests } from './block-rte/manifests.js';
 import { manifests as blockTypeManifests } from './block-type/manifests.js';
 import type { ManifestTypes } from '@umbraco-cms/backoffice/extension-registry';
+// TODO: Remove test custom view, or transfer to test or similar?
```
Comment (iOvergaard) [inline src/packages/block/manifests.ts:7]: So we have overcome this before by adding custom manifests to the manifest MSW handler, which you can then turn on by setting the `VITE_UMBRACO_EXTENSION_MOCKS` to `on`. I would recommend you move this code to that handler. That way you can always turn it "on" by a simple environment value.

**Review by iOvergaard — CHANGES_REQUESTED** [review]
I can get a custom view to show. I get all of the content of the block served to my custom element. So the feature looks good.

What I severely missed was a way to display an image in my custom view; I mean, I get the key and everything, but converting that from a GUID into a usable URL was harder than I imagined. I guess we could do some good with a few helper functions in this area. Anyway, that doesn't affect the feature as such.

I left a few comments of things I stumbled upon:

File: src/packages/block/block-grid/components/block-grid-entry/block-grid-entry.element.ts [hunk 2]
```diff
@@ -225,6 +305,19 @@ export class UmbBlockGridEntryElement extends UmbLitElement implements UmbProper
...
+		}
+		if (manifest.forBlockEditor && !stringOrStringArrayContains(manifest.forBlockEditor, UMB_BLOCK_GRID)) {
+			return false;
+		}
+		return true;
+	};
+
 	#renderInlineEditBlock() {
 		return html`<umb-block-grid-block-inline
 			.contentUdi=${this.contentUdi}
```
[the remark this answers is not in the document]
Comment (nielslyngsoe) [reply src/packages/block/block-grid/components/block-grid-entry/block-grid-entry.element.ts]: your right

**Review by iOvergaard — APPROVED** [review]
LGTM. We can always move the test custom view into the mock handlers in another branch.

**Pull request merged**